src/fontra/views/editor/panel-designspace-navigation.js

```diff
@@ -103,6 +103,216 @@ export default class DesignspaceNavigationPanel extends Panel {
       () => this._updateResetAllAxesButtonState(),
       100
     );
+
+    this.fontController.ensureInitialized.then(() => {
```

Comment:
This change is too big. Also, the body of this arrow function is too big.

Can you try to do this:
- revert the `setup()` method change
- _call_ `.setup()` from a "then" handler

Since setup is big, I would like to be sure nothing important changes there. By inlining the whole thing in an arrow function I have no way to compare before and after.

Reply:
If the name "setup" is problematic (as you seem to say in https://github.com/googlefonts/fontra/pull/1036#issuecomment-1866883562), then please suggest another name, instead of deleting the entire method.

Reply:
I reverted the commit, kept it as `setup`, I suggest to break the method into several pieces as a next step. As you say, this method is too big. Only updateAxes, updateSources calls at the end requires fontController to be initialized. We can move the event bindings to the constructor. Create another method named `createUIList`. I am not sure if we should do that in this PR.